pkg/proc/fncall.go

```diff
@@ -763,8 +823,14 @@ func funcCallStep(callScope *EvalScope, fncall *functionCallState, thread Thread
 		cfa := regs.SP()
 		oldpc := regs.PC()
 		callOP(bi, thread, regs, fncall.fn.Entry)
+		formalScope, err := GoroutineScope(callScope.target, thread)
+		if formalScope != nil && formalScope.Regs.CFA != int64(cfa) {
+			err = fmt.Errorf("mismatch in CFA %#x (calculated) %#x (expected)", formalScope.Regs.CFA, int64(cfa))
```

Comment:
Can you add some comments around this for context and why we might find ourselves in this situation.

Reply:
It should never happen, I'll add a comment to that effect.